internal/state/store.go

```diff
@@ -420,6 +427,14 @@ func (store dbStore) PruneStates(from int64, to int64, evidenceThresholdHeight i
 		return err
 	}
 
+	// We do not want to panic or interrupt consensus on compaction failure
+	if store.StoreOptions.Compact && (to%store.StoreOptions.CompactionInterval == 0 || pruned >= uint64(store.StoreOptions.CompactionInterval)) {
+		err = store.db.Compact(nil, nil)
+		if err != nil && store.StoreOptions.Logger != nil {
+			store.StoreOptions.Logger.Error("error compacting state store ", "err", err)
```

Comment:
I think it should be OK to return an error here, because this will bubble up to `pruneBlocksToRetainHeight` in the `pruner.go` and there an error is logged too afaiu.

Reply:
Ah yes you are right. This was done based on the 0.37 backport where an error here actually caused a panic eventually. This can be indeed returned then.